### ISSUE TITLE
RichMenu

### ISSUE THREAD

**Issue by riri93** [line/line-bot-sdk-java#168]
How can I create a menu for LINE with Java code? is the RichMenu constructor not available? because I created a RichMenuBounds, RichMenuSize, RichMenuArea But I couldn't create a RichMenu. Can you please help me with this isuue? Thank you

**Comment by kazuki-ma** [comment]
Here is a example.
```java
    private final LineMessagingClient client;

    void createRichMenu() {
        final RichMenu richmenu = RichMenu
                .builder()
                .name("NAME")
                .areas(richMenuAreas())
                .chatBarText("CHAT_BAR_TEXT")
                .selected(true)
                .size(RichMenuSize.FULL)
                .build();

       client.createRichMenu(richmenu);
    }

    private static List<RichMenuArea> richMenuAreas() {
        return asList(richMenuArea(0, 0, "A"),
                      richMenuArea(1, 0, "B"),
                      richMenuArea(2, 0, "C"),
                      richMenuArea(0, 1, "D"),
                      richMenuArea(1, 1, "E"),
                      richMenuArea(2, 1, "F"));
    }

    private static RichMenuArea richMenuArea(final int xn, final int yn, String name) {
        final int X_STEP = 833;
        final int Y_STEP = 843;

        return new RichMenuArea(new RichMenuBounds(X_STEP * xn, Y_STEP * yn, X_STEP, Y_STEP),
                                new MessageAction("", name + " is tapped"));
    }
```

For `RichMenu`, there is no default constructor please call `RichMenu.builder()` first, put attributes and finally call `build()` method.

After that, you will get following log.

```
2018-01-06 23:46:59.170  INFO 40159 --- [api.line.me/...] com.linecorp.bot.client.wire             
: --> POST https://api.line.me/v2/bot/richmenu http/1.1
: Content-Type: application/json; charset=UTF-8
: Content-Length: 794
: Authorization: Bearer QHfxxxxxxx==
: User-Agent: line-botsdk-java/version
: 
: {"size":{"width":2500,"height":1686},"selected":true,"name":"type_name_here","chatBarText":"type_name_here","areas":[{"bounds":{"x":0,"y":0,"width":833,"height":843},"action":{"type":"message","label":"","text":"A is tapped"}},{"bounds":{"x":833,"y":0,"width":833,"height":843},"action":{"type":"message","label":"","text":"B is tapped"}},{"bounds":{"x":1666,"y":0,"width":833,"height":843},"action":{"type":"message","label":"","text":"C is tapped"}},{"bounds":{"x":0,"y":843,"width":833,"height":843},"action":{"type":"message","label":"","text":"D is tapped"}},{"bounds":{"x":833,"y":843,"width":833,"height":843},"action":{"type":"message","label":"","text":"E is tapped"}},{"bounds":{"x":1666,"y":843,"width":833,"height":843},"action":{"type":"message","label":"","text":"F is tapped"}}]}
: --> END POST (794-byte body)
: <-- 200  https://api.line.me/v2/bot/richmenu (228ms)
: Server: nginx
: Date: Sat, 06 Jan 2018 14:46:59 GMT
: Content-Type: application/json;charset=UTF-8
: Transfer-Encoding: chunked
: Connection: keep-alive
: X-Line-Request-Id: ad832019-8de0-4031-aa62-9c1e2301ecb9
: X-Content-Type-Options: nosniff
: X-XSS-Protection: 1; mode=block
: Cache-Control: no-cache, no-store, max-age=0, must-revalidate
: Pragma: no-cache
: Expires: 0
: X-Frame-Options: DENY
: 
: {"richMenuId":"richmenu-afce0e6a1aae964c643711f3a8623efa"}
: <-- END HTTP (58-byte body)
```

**Comment by riri93** [comment]
Thank you for your answer. 
For this part: `client.createRichMenu(richmenu);`  it says "The method createRichMenu(RichMenu) is undefined for the type LineMessagingClient"
Can you tell me what is the problem? is it a Line version problem?
And how can I create or edit a menu for a specific user?

**Comment by kazuki-ma** [comment]
Do you use the latest version? if didn't, please update.
RichMenues related methods are available since 1.12.

**Comment by kazuki-ma** [comment]
Let me close the issue. If you have problem, please reopen it.
